### PR TITLE
CBAPI-2486: Live response guide and migration guide

### PR DESCRIPTION
**Pull request by emitreva-cb**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Tests have been added that prove the fix is effective or that the feature works.
- [ ] New and existing tests pass locally with the changes.
- [ ] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [ ] A self-review of the code has been done.

## What is the ticket or issue number?
https://jira.carbonblack.local/browse/CBAPI-2486

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: docs/authentication.rst [hunk 1]
```diff
@@ -16,11 +16,9 @@ specific actions, and use them as needed.
 For example, if using the
 `Platform Devices API <https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/devices-api/#search-devices>`_
 to search for mission critical devices, and the
-`Endpoint Standard Live Response API <https://developer.carbonblack.com/reference/carbon-black-cloud/cb-defense/latest/live-response-api/>`_
-to execute commands on those devices, generate two API Keys. The Platform API Key should have the
-Custom Access Level, and the Live Response Key should have the Live Response Access Level.
-Store these Keys with profile names, and reference the profile names when creating
-CBCloudAPI objects.
+`Platform Live Response API <https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/live-response-api/>`_
+to execute commands on those devices, generate two API Keys with Custom Access Level with appropriate permissions.
```
Comment (emitreva-cb) [inline docs/authentication.rst:20]: Not sure whether this should be changed to say one key and explain that the same key can be used for Live Response and querying devices. Any suggestions?

File: docs/live-response.rst [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+Note that we need to pass the pid of the process to kill it.
+
+Additional Resources
+--------------------
+
+There are a number other commands supported. For full list and description see
+`the developer documentation <https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/live-response-api/>`_
+for details.
+
+Check the migration guide of Live Response from v3 to v6 :doc:`live-response-v6-migration`
```
Comment (emitreva-cb) [inline docs/live-response.rst:84]: This is the only place I have put the link to the migration guide - not sure whether it should be placed somewhere else.

File: docs/live-response.rst [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    >>> api = CBCloudAPI(profile='sample')
+    >>> from cbc_sdk.platform import Device
+    >>> device = api.select(Device).first()
+    >>> lr_session = device.lr_session()
+
+File Commands
+-------------
+
+Once a session is established we are going to create a directory and upload a file in that newly created directory.
+List directory command is returning the content of the directory including the uploaded file.
```
Comment (lhilmes-cb) [inline docs/live-response.rst:24]: This read a little odd to me. Consider changing to: "The `List directory` command returns the content of the directory, including the uploaded file." 
(I'm not sure how `List directory` should be formatted though.)

File: docs/live-response.rst [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    >>> lr_session.put_file(open("demo.txt", "r"), 'C:\\\\demo\\\\demo.txt')
+    >>> directories = lr_session.list_directory('C:\\\\demo\\\\')
+    >>> for directory in directories:
+    ...   print(f"{directory['attributes'][0]} {directory['filename']}")
+    ...
+    DIRECTORY .
+    DIRECTORY ..
+    ARCHIVE demo.txt
+
+Note that the creation of the directory will fail if the directory already exists.
```
Comment (lhilmes-cb) [inline docs/live-response.rst:38]: often for notes we use italics:

_Note: the creation of a directory will fail if the directory already exists._

File: docs/live-response.rst [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+Now we will get the contents of the file and then delete the file and the directory.
```
Comment (lhilmes-cb) [inline docs/live-response.rst:40]: Consider changing to:
"Next, get the contents of the file and then delete the file and the directory."

File: docs/live-response.rst [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+::
+
+    >>> contents = lr_session.get_file('C:\\\\demo\\\\demo.txt')
+    >>> lr_session.delete_file('C:\\\\demo\\\\demo.txt')
+    >>> lr_session.delete_file('C:\\\\demo\\\\')
+
+Note that we could delete a directory with the delete file command.
```
Comment (lhilmes-cb) [inline docs/live-response.rst:48]: Change to:
_Note: you can also delete a directory with the `delete file` command._

File: docs/live-response.rst [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+
+Process Commands
+----------------
+We can also execute commands to manage processes. Once we have established a session we could check the
```
Comment (lhilmes-cb) [inline docs/live-response.rst:52]: Change to:
"You can also execute commands to manage processes. Once you have established a session, you can check running processes."

File: docs/live-response.rst [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+::
+
+    >>> processes = lr_session.list_processes()
+    >>> for process in processes:
+    ...   print(f"{process['process_pid']} {process['process_path']}")
+    ...
+    42 c:\windows\explorer.exe
+    43 c:\windows\system32\svchost.exe
+
+We can create a process and kill it.
```
Comment (lhilmes-cb) [inline docs/live-response.rst:64]: Update to: "You can also create or kill a process."

File: docs/live-response.rst [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+::
+
+    >>> lr_session.create_process(r'cmd.exe /c "ping.exe -t 127.0.0.1"',
+                                  wait_for_completion=False, wait_for_output=False)
+    >>> processes = lr_session.list_processes()
+    >>> for process in processes:
+    ...     if 'ping.exe' in process['process_path']:
+    ...         lr_session.kill_process(process['process_pid'])
+
+Note that we need to pass the pid of the process to kill it.
```
Comment (lhilmes-cb) [inline docs/live-response.rst:75]: Change to:
_Note: you must pass the PID of the process to kill it._

File: docs/live-response.rst [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+Additional Resources
+--------------------
+
+There are a number other commands supported. For full list and description see
```
Comment (lhilmes-cb) [inline docs/live-response.rst:80]: Change to:
"Find a full list of supported commands in the `Live Response API documentation <https://developer.carbonblack.com/reference/carbon-black-cloud/platform/latest/live-response-api/>`_.

File: docs/live-response.rst (hunk 2 above)
Comment (lhilmes-cb) [inline docs/live-response.rst:84]: Consider changing to "For tips on migrating from Live Response v3 to v6, check the `migration guide`."
(not sure how that link should be formatted)

**Review by avanbrunt-cb — APPROVED** [review]

**Review by lhilmes-cb — APPROVED** [review]
Thanks for making those changes! Looks great.

**Comment by emitreva-cb** [comment]
> Thanks for making those changes! Looks great.

Thank you for all the help. Lisa!

**Pull request merged**